pkgs/build-support/dlang/builddubpackage/default.nix

```diff
@@ -0,0 +1,73 @@
+{ lib
+, stdenv
+, makeDubDep
+, dub
+, ldc
+, removeReferencesTo
+}:
+
+# TODO: create proper documentation
+# until then you can ping @TomaSajt if you have any questions
+
+{ dubDeps
+, extraDubDeps ? [ ]
+, dubBuildType ? "release"
+, dubFlags ? [ ]
+, compiler ? ldc
+, ...
+}@args:
+
+let
+  combinedDeps = (import dubDeps { inherit makeDubDep; }) ++ extraDubDeps;
+in
+stdenv.mkDerivation (builtins.removeAttrs args [ "dubDeps" "extraDubDeps" ] // {
+  nativeBuildInputs = args.nativeBuildInputs or [ ] ++ [
+    dub
+    compiler
+    removeReferencesTo
+  ];
+
+  postUnpack = ''
+    # the dependencies need to be placed into a deterministic location, because
+    # the source file paths are included in the final binaries
+    export DUB_DEPS="$NIX_BUILD_TOP/.dub-deps"
```

Comment:
I don't think NIX_BUILD_TOP is deterministic. Doing it this way is still better than nothing though.

Reply:
Can it be set to something other than `/build`? (without changing the drv hash)